cuqi/experimental/mcmc/_cwmh.py

```diff
@@ -0,0 +1,207 @@
+import numpy as np
+import cuqi
+from cuqi.experimental.mcmc import ProposalBasedSamplerNew
+from cuqi.array import CUQIarray
+from numbers import Number
+
+class CWMHNew(ProposalBasedSamplerNew):
+    """Component-wise Metropolis Hastings sampler.
+
+    Allows sampling of a target distribution by a component-wise random-walk
+    sampling of a proposal distribution along with an accept/reject step.
+
+    Parameters
+    ----------
+
+    target : `cuqi.distribution.Distribution` or lambda function
+        The target distribution to sample. Custom logpdfs are supported by using
+        a :class:`cuqi.distribution.UserDefinedDistribution`.
+    
+    proposal : `cuqi.distribution.Distribution` or callable method
+        The proposal to sample from. If a callable method it should provide a
+        single independent sample from proposal distribution. Defaults to a
+        Gaussian proposal.  *Optional*.
+
+    scale : float or ndarray
+        Scale parameter used to define correlation between previous and proposed
+        sample in random-walk.  *Optional*. If float, the same scale is used for
+        all dimensions. If ndarray, a (possibly) different scale is used for
+        each dimension.
+
+    initial_point : ndarray
+        Initial parameters. *Optional*
+
+    dim : int
```

Comment:
I would suggest we remove the dim property. Instead user can define UserDefinedDistribution. We can discuss at next meeting.

Reply:
After our discussion, we agreed to remove the "dim" property yes.